### PR TITLE
Serving front end

### PR DESCRIPTION
**Pull request by glorysdj**
This change is [Reviewable](https://reviewable.io/reviews/intel-analytics/analytics-zoo/2301)

### CODE REVIEW

File: zoo/pom.xml [hunk 1]
```diff
@@ -106,6 +107,69 @@
             </properties>
         </profile>
 
+        <profile>
+            <id>serving-frontend</id>
```
Comment (jason-dai) [inline zoo/pom.xml:111]: serving-http

File: zoo/src/main/scala/com/intel/analytics/zoo/serving/frontend/FrontEndApp.scala [hunk 2]
```diff
@@ -0,0 +1,159 @@
+/*
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/serving/frontend/FrontEndApp.scala:1]: zoo/serving/http/

File: zoo/pom.xml (hunk 1 above)
Comment (glorysdj) [reply zoo/pom.xml:111]: ok, will change name to http

File: zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  override val logger = LoggerFactory.getLogger(getClass)
+
+  val name = "analytics zoo web serving frontend"
+
+  implicit val system = ActorSystem("zoo-serving-frontend-system")
+  implicit val materializer = ActorMaterializer()
+  implicit val executionContext = system.dispatcher
+  implicit val timeout: Timeout = Timeout(100, TimeUnit.SECONDS)
+
+  val redisInputQueue = "image_stream"
```
Comment (Litchilitchy) [inline zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala:46]: we are using "serving_stream" this single stream name to be the queue of all inputs including image, tensor, sparse tensor now

File: zoo/src/test/scala/com/intel/analytics/zoo/serving/FrontendActorsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+import com.intel.analytics.zoo.serving.http.{PredictionInputMessage, _}
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+import redis.clients.jedis.Jedis
+
+import scala.concurrent.{Await, ExecutionContextExecutor}
+import scala.util.Random
+
+class FrontendActorsSpec extends FlatSpec with Matchers with BeforeAndAfter with Supportive {
+  val random = new Random()
+  val redisHost = "localhost"
```
Comment (Litchilitchy) [inline zoo/src/test/scala/com/intel/analytics/zoo/serving/FrontendActorsSpec.scala:36]: Should also use our deployed server on gateway?

Comment (glorysdj) [reply zoo/src/test/scala/com/intel/analytics/zoo/serving/FrontendActorsSpec.scala:36]: this is ut, no need to have dependency to a real redis server.

File: zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala (hunk 3 above)
Comment (jason-dai) [reply zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala:46]: > we are using "serving_stream" this single stream name to be the queue of all inputs including image, tensor, sparse tensor now

When will this change be ready?

Comment (glorysdj) [reply zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala:46]: i will change to serving_stream.
the frontend  code is ready to process json inputs for structured data, i will work with jiaming to also make it working at serving backend.

File: zoo/src/main/scala/com/intel/analytics/zoo/serving/http/domains.scala [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+      val shape = key_field._2._1
+      val data = key_field._2._2
+      val (isList, fieldSampe) = data.isInstanceOf[ArrayBuffer[_]] match {
+        case true => (true, data.asInstanceOf[ArrayBuffer[_]](0))
+        case false => (false, data)
+      }
+      if (fieldSampe.isInstanceOf[Int]) {
+        val field = if (isList) {
+          val shapeSize = shape.asInstanceOf[ArrayBuffer[_]].size
+          val shapeList = new util.ArrayList[Field]()
```
Comment (cyita) [inline zoo/src/main/scala/com/intel/analytics/zoo/serving/http/domains.scala:123]: `shapeSize`, `shapeList` and `dataSize` are not needed in PR #2347

File: zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala (hunk 3 above)
Comment (Litchilitchy) [reply zoo/src/main/scala/com/intel/analytics/zoo/serving/http/FrontEndApp.scala:46]: > > we are using "serving_stream" this single stream name to be the queue of all inputs including image, tensor, sparse tensor now
> 
> When will this change be ready?

https://github.com/intel-analytics/analytics-zoo/pull/2347

File: zoo/src/main/scala/com/intel/analytics/zoo/serving/http/domains.scala (hunk 5 above)
Comment (glorysdj) [reply zoo/src/main/scala/com/intel/analytics/zoo/serving/http/domains.scala:123]: i think we can use class instances here to handle all the structured data, scalar, tensor and sparsetensor.

**Comment by glorysdj** [comment]
jenkins passed

**Comment by jason-dai** [comment]
Please add user doc

**Pull request merged**